diff --git a/configure.ac b/configure.ac
index 020cd8c..851ade5 100644
--- a/configure.ac
+++ b/configure.ac
@@ -3233,7 +3233,7 @@ AS_IF([test "$with_dln_a_out" != yes], [
 	[linux* | gnu* | k*bsd*-gnu | netbsd* | bsdi* | kopensolaris*-gnu | haiku*], [
 			: ${LDSHARED='$(CC) -shared'}
 			AS_IF([test "$rb_cv_binary_elf" = yes], [
-			    LDFLAGS="$LDFLAGS -Wl,-export-dynamic"
+			    LDFLAGS="$LDFLAGS -Wl,-export-dynamic -Wl,--no-undefined"
 			])
 			rb_cv_dlopen=yes],
 	[interix*], [	: ${LDSHARED='$(CC) -shared'}
