diff -up ruby-2.7.4/ext/openssl/ossl_pkey_rsa.c.omv~ ruby-2.7.4/ext/openssl/ossl_pkey_rsa.c
--- ruby-2.7.4/ext/openssl/ossl_pkey_rsa.c.omv~	2021-09-11 02:54:42.145240057 +0200
+++ ruby-2.7.4/ext/openssl/ossl_pkey_rsa.c	2021-09-11 02:54:52.066213379 +0200
@@ -939,7 +939,6 @@ Init_ossl_rsa(void)
     rb_define_method(cRSA, "params", ossl_rsa_get_params, 0);
 
     DefRSAConst(PKCS1_PADDING);
-    DefRSAConst(SSLV23_PADDING);
     DefRSAConst(NO_PADDING);
     DefRSAConst(PKCS1_OAEP_PADDING);
 
