--- ruby-1.8.7.p352/configure.in.orig	2011-09-16 20:27:10.537941882 -0300
+++ ruby-1.8.7.p352/configure.in	2011-09-16 20:27:15.879970708 -0300
@@ -86,7 +86,7 @@ if test "$program_prefix" = NONE; then
   program_prefix=
 fi
 AC_CANONICAL_TARGET
-target_os=`echo $target_os | sed 's/linux-gnu$/linux/;s/linux-gnu/linux-/'`
+target_os=`echo $target_os | sed 's/linux-gnu\(eabi\)\?$/linux/;s/linux-gnu\(eabi\)\?/linux-/'`
 ac_install_sh='' # unusable for extension libraries.
 
 ifelse(currently,disabled, [
