--- ruby-1.9.3-p0/lib/mkmf.rb.shared~	2012-01-18 19:15:17.000000000 +0100
+++ ruby-1.9.3-p0/lib/mkmf.rb	2012-01-18 19:16:56.307910394 +0100
@@ -396,7 +396,7 @@ MSG
 end
 
 def link_command(ldflags, opt="", libpath=$DEFLIBPATH|$LIBPATH)
-  librubyarg = $extmk ? $LIBRUBYARG_STATIC : "$(LIBRUBYARG)"
+  librubyarg = $extmk ? $LIBRUBYARG_SHARED : "$(LIBRUBYARG)"
   conf = RbConfig::CONFIG.merge('hdrdir' => $hdrdir.quote,
                                 'src' => "#{CONFTEST_C}",
                                 'arch_hdrdir' => $arch_hdrdir.quote,
