--- ruby-2.0.0-p451/tool/rbinstall.rb~	2014-03-09 23:47:10.797069948 +0000
+++ ruby-2.0.0-p451/tool/rbinstall.rb	2014-03-10 00:45:47.642333269 +0000
@@ -712,7 +712,7 @@
 install?(:ext, :comm, :gem) do
   $:.unshift(File.join(srcdir, "lib"))
   require("rubygems.rb")
-  gem_dir = Gem.default_dir
+  gem_dir = Gem.default_dirs[:local][:gem_dir]
   # Gem.ensure_gem_subdirectories makes subdirectories group-writable.
   directories = Gem::REPOSITORY_SUBDIRECTORIES
   prepare "default gems", gem_dir, directories
